Assets/BossRoom/Scripts/Server/Game/Action/MeleeAction.cs

```diff
@@ -64,36 +64,47 @@ public override bool Update()
             return true;
         }
 
-
         /// <summary>
         /// Returns the ServerCharacter of the foe we hit, or null if none found.
         /// </summary>
         /// <returns></returns>
         private IDamageable DetectFoe(ulong foeHint = 0)
         {
-            //this simple detect just does a boxcast out from our position in the direction we're facing, out to the range of the attack.
+            return GetIdealMeleeFoe(Description.IsFriendly ^ m_Parent.IsNpc, m_Parent.GetComponent<Collider>(), Description.Range, foeHint);
+        }
 
+        /// <summary>
+        /// Utility used by Actions to perform Melee attacks. Performs a melee hit-test
+        /// and then looks through the results to find an alive target, preferring the provided
+        /// enemy.
+        /// </summary>
+        /// <param name="isNPC">true if the attacker is an NPC (and therefore should hit PCs). False for the reverse.</param>
+        /// <param name="attacker">The collider of the attacking GameObject.</param>
+        /// <param name="range">The range in meters to check for foes.</param>
+        /// <param name="preferredTargetNetworkId">The NetworkObjectId of our preferred foe, or 0 if no preference</param>
+        /// <returns></returns>
+        public static IDamageable GetIdealMeleeFoe(bool isNPC, Collider ourCollider, float meleeRange, ulong preferredTargetNetworkId)
```

Comment:
Make sure all params are included in the summary params and are contextual.

Reply:
ok